### PR TITLE
update for rh5/rh6 compatability

### PR DESCRIPTION
**Pull request by sparciii**
nrpe_program method should be set to :program attribute provided by NrpeService rather than hardcoded entry.

override sysvinit (rhel5) service template to not specify pid as that is handled via nrpe.cfg. use default config provided by nrpe package. 
override upstart (rhel6) service template to use `expect fork`.

this gets the nrpe service running and restart-able via appropriate init system.

### CODE REVIEW

File: libraries/nrpe_installation_package.rb [hunk 1]
```diff
@@ -126,7 +126,7 @@ def nagios_plugins
       # @return [String]
       # @api private
       def nrpe_program
-        options.fetch(:program, '/usr/sbin/nrpe')
+        options.fetch(:program, @program)
```
Comment (johnbellone) [inline libraries/nrpe_installation_package.rb]: Where are we storing that instance variable? The second parameter is the default value.

File: libraries/nrpe_installation_archive.rb [hunk 2]
```diff
@@ -95,7 +95,7 @@ def nagios_plugins
       # @return [String]
       # @api private
       def nrpe_program
-        options.fetch(:program, ::File.join(nrpe_base, 'bin', 'nrpe'))
+        options.fetch(:program, @program)
```
Comment (johnbellone) [inline libraries/nrpe_installation_archive.rb]: Where are we storing that instance variable? The second parameter is the default value.

**Comment by johnbellone** [comment]
Other than those two comments 👍

**Comment by johnbellone** [comment]
These options are set slightly different... they the nrpe-ng namespace in attributes and are nested under the name of the resource. By default thats _nrpe_ so this would work, but if the name of the resource was foo it would not.

``` ruby
node['nrpe-ng']['nrpe']['program']
```

**Comment by sparciii** [comment]
@johnbellone that worked, thanks.

**Pull request merged**